zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/archiver/Archiver.java

```diff
@@ -142,9 +148,11 @@ private static BatchOperationArchiverJob createBatchOperationJob(
         executor);
   }
 
-  private static ScheduledThreadPoolExecutor defaultExecutor(final ThreadFactory threadFactory) {
-    // TODO: set size to 2 in case we need to do batch operations
-    final var executor = new ScheduledThreadPoolExecutor(1, threadFactory);
+  private static ScheduledThreadPoolExecutor defaultExecutor(
+      final ThreadFactory threadFactory, final int partitionId) {
+    // if we are present on partition 1, we need to have 2 threads to handle both jobs
+    final var executor =
+        new ScheduledThreadPoolExecutor(partitionId == START_PARTITION_ID ? 2 : 1, threadFactory);
```

Comment:
:wrench: We now have the check for `partitionId == START_PARTITION_ID` duplicated. What if we ever change the logic? How about:

- We create the jobs in a list/array. 
- The batch operation job is added only if the `partitionId == START_PARTITION_ID`.
- We pass the number of jobs to the default executor method, and we create `Math.max(jobs.size(), BG_THREAD_LIMIT)` (where `BG_THREAD_LIMIT` is some sane value - we can add a TODO about making it configurable in the future).

That way the logic is more or less in one place, and when we add more jobs, we will automatically "parallelize" them.

Reply:
Does not seem super easy to do since we have to instantiate the executor before we create the jobs which also needs to know the number of jobs 🤔 

Later, we are going to add the post archiver, which will run in all partitions, so it will not add to logic complexity from here. However, for additional jobs, I am not sure if a different logic is used. 

Do you think it is ok to merge as is now, and after the post archiver jobs decide on the limits of the number of threads and if we should use different checks here?

Reply:
Let's do that.